frontends/common/options.cpp

```diff
@@ -96,6 +96,10 @@ CompilerOptions::CompilerOptions() : Util::Options(defaultMessage) {
     registerOption("--p4runtime-file", "file",
                    [this](const char* arg) { p4RuntimeFile = arg; return true; },
                    "Write a P4Runtime control plane API description to the specified file.");
+    registerOption("--p4runtime-entries-file", "file",
```

Comment:
I think that this should be part of the BMV2Options class instead.
Other front-ends to do not have this capability.

Reply:
I disagree, this should be common to all backends
In terms of applicability, this is no different from `--p4runtime-file`

Reply:
Agreed with Antonin. Keep in mind that while we're not there yet, we want to get to the point where the P4Runtime serializer isn't backend-specific at all; it should only require the frontend to be run.

Reply:
You are assuming that all back-ends need the same control-plane, this will not be true.
A switch like OVS will have a separate control-plane.
If you think it is common to all back-ends you should explain how the ebpf back-end can benefit.

Reply:
That's, as I see it, a separate question. A P4Info message (which is what the P4Runtime serializer code, as it currently stands, really generates) just describes metadata about the program. That information is intended to be used by the control plane, but even if people using a certain backend or target will end up using a different control plane scheme, we can still *generate* a P4Info message. It's still well-defined.

Reply:
In other words, nobody has to use it, and some people won't, but there's no real advantage to not providing it for certain backends and making things uniform adds convenience and makes the ecosystem simpler.